datafusion/core/src/physical_plan/joins/hash_join.rs

```diff
@@ -757,7 +757,7 @@ pub fn build_equal_condition_join_indices(
     let mut build_indices = UInt64BufferBuilder::new(0);
     let mut probe_indices = UInt32BufferBuilder::new(0);
     // Visit all of the probe rows
-    for (row, hash_value) in hash_values.iter().enumerate() {
+    for (row, hash_value) in hash_values.iter().enumerate().rev() {
```

Comment:
Why does this need to be reverted as well?

Reply:
Ah because otherwise the order of is wrong - I get it. Could we add some comments for both changes?

Reply:
Sure, I am on it.

Reply:
Sorry for this issue https://github.com/apache/arrow-datafusion/issues/7113
I don't know why this issue is a bug?
Does it get the error result for the hash join?

@Dandandan @metesynnada

Reply:
So there seems there was no test for checking the ordering of the hash join results (only for the matching rows to be correct) @metesynnada changed one to test this. This is important if the execution plan depends on the output ordering (and doesn't add extra sort).

The matching rows themselves are correct even before this PR.